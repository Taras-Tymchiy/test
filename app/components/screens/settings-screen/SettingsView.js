// @flow
import * as React from 'react';
import {
  Platform,
  Text,
  TextInput,
  View
} from 'react-native';
import FormControlWrapper from '../../molecules/FormControlWrapper';
import styles from './SettingsView.styles';

type SettingsViewState = {
  syncInterval: string;
  postsUrl: string;
  postsCount: string;
}

export type SettingsViewProps = {
  syncInterval: number;
  postsUrl: string;
  postsCount: number;
  intervalChanged: (v: number) => void;
  urlChanged: (v: string) => void;
  postsCountChanged: (v: number) => void;
}

export default class SettingsView 
  extends React.Component<SettingsViewProps, SettingsViewState> {
    constructor(props: SettingsViewProps) {
      super(props);
      this.state = this.propsToState(props);
    }

    propsToState({ syncInterval, postsUrl, postsCount }: SettingsViewProps) {
      return  { 
        postsUrl, 
        syncInterval: String(syncInterval),
        postsCount: String(postsCount)
      };
    }

    componentWillReceiveProps(newProps: SettingsViewProps) {
      this.setState(this.propsToState(newProps));
    }

    toNumeric = (val: string)=> String(val).replace(/[^0-9]/g, ''); 

    onIntervalChange = (v: string) => { this.setState({syncInterval: this.toNumeric(v)}) };
    onIntervalSave = () => { this.props.intervalChanged(+this.state.syncInterval) };

    onPostsCountChange = (v: string) => { this.setState({postsCount: this.toNumeric(v)}) };
    onPostsCountSave = () => { this.props.postsCountChanged(+(this.state.postsCount || 0)) };
    
    onUrlChange = (postsUrl: string) => { this.setState({postsUrl}) };
    onUrlSave = () => { this.props.urlChanged(this.state.postsUrl) };
    
    onDiscard = () => {
      this.setState(this.propsToState(this.props)); 
    };
    
    render () {
      const { syncInterval, postsUrl, postsCount } = this.state;
      return (
        <View style={styles.container} >
          {/* $FlowFixMe */}
          <FormControlWrapper 
              title="Endpoint url"
              value={postsUrl} 
              onSave={this.onUrlSave} 
              onDiscard={this.onDiscard} >
            <TextInput value={postsUrl} multiline={true} onChangeText={this.onUrlChange} />
          </FormControlWrapper>
          <FormControlWrapper 
              title="Posts number"
              description="The number of posts to fetch from API and display (0 means unlimited)"
              value={postsCount} 
              onSave={this.onPostsCountSave}
              keyboardType = 'numeric'
              onDiscard={this.onDiscard} >
            <TextInput value={postsCount} onChangeText={this.onPostsCountChange} />
          </FormControlWrapper>
          <FormControlWrapper 
              title="Sync interval" 
              description="Synchronization interval in milliseconds. Values less than 1000 are not allowed and will be overwritten to 1000"
              value={`${syncInterval} ms`} 
              onSave={this.onIntervalSave} 
              keyboardType = 'numeric'
              onDiscard={this.onDiscard} >
            <TextInput value={syncInterval} onChangeText={this.onIntervalChange} />
          </FormControlWrapper>
        </View>
      );
    }
};
