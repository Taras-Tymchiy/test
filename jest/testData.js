import { Post } from "../app/entities/Post";


export const posts: Post[] = [
    {
        id: 1,
        id_str: '1',
        created_at: '2015-01-01',
        user: {
            id: 1,
            name: 'John Snow',
        },
        text: 'post text 1'
    },
    {
        id: 2,
        id_str: '2',
        created_at: '2015-01-02',
        user: {
            id: 2,
            name: 'Barack Obama',
        },
        text: 'post text 2'
    }
]