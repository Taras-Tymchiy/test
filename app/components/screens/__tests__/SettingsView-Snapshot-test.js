import React from 'react';
import renderer from 'react-test-renderer';
import SettingsView from '../settings-screen/SettingsView';
import { posts } from '../../../../jest/testData';

jest.mock('../../molecules/FormControlWrapper', (props) => `FormControlWrapper ${props}`);

const props = {
  pullIntervl: 5000,
  postsCount: 10,
  postsUrl: 'test',
  intervalChanged: () => null,
  urlChanged: () => null,
  postsCountChanged: () => null
};

it('renders posts in a SettingsView using Snapshots', () => {
  expect(renderer.create(
    <SettingsView {...props} />
  )).toMatchSnapshot();
});
