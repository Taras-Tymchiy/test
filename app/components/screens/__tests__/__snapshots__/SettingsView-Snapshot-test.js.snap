// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders posts in a SettingsView using Snapshots 1`] = `
<View
  style={
    Object {
      "borderColor": "#CCC",
      "borderTopWidth": 0.5,
      "marginBottom": 24,
      "marginTop": 24,
    }
  }
>
  <FormControlWrapper undefined
    onDiscard={[Function]}
    onSave={[Function]}
    title="Endpoint url"
    value="test"
  >
    <TextInput
      allowFontScaling={true}
      multiline={true}
      onChangeText={[Function]}
      value="test"
    />
  </FormControlWrapper undefined>
  <FormControlWrapper undefined
    description="The number of posts to fetch from API and display (0 means unlimited)"
    keyboardType="numeric"
    onDiscard={[Function]}
    onSave={[Function]}
    title="Posts number"
    value="10"
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      value="10"
    />
  </FormControlWrapper undefined>
  <FormControlWrapper undefined
    description="Synchronization interval in milliseconds. Values less than 1000 are not allowed and will be overwritten to 1000"
    keyboardType="numeric"
    onDiscard={[Function]}
    onSave={[Function]}
    title="Sync interval"
    value="undefined ms"
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      value="undefined"
    />
  </FormControlWrapper undefined>
</View>
`;
